common/functions/src/scalars/strings/string.rs

```diff
@@ -25,5 +30,10 @@ impl StringFunction {
         factory.register("substring", SubstringFunction::desc());
         factory.register("oct", OctFunction::desc());
         factory.register("repeat", RepeatFunction::desc());
+        factory.register("ltrim", LTrimFunction::desc());
```

Comment:
Great,  I have an idea about these functions.

They have the same return_type and number of arguments and datatypes, the difference between them are `X` where `string_array.append_option(X)`, so it's better to have a generic function to reduce the code size?

Refer: https://github.com/datafuselabs/databend/blob/main/common/functions/src/scalars/dates/number_function.rs#L475-L493